@pnpm/cli-utils is unable to resolve pnpm's package.json when using self-installer

<!--
**Please only file bugs/feature requests for pnpm here.**

If you have a question that is not answered in the FAQ section of the README
then it is best to post your question on StackOverflow under the pnpm tag:
https://stackoverflow.com/questions/tagged/pnpm
or ask it in our Gitter chatroom:
https://gitter.im/pnpm/pnpm

If your issue is a bug, please follow the format below:
-->

I've done some investigation into the issue, and I think it stems from this change: https://github.com/pnpm/pnpm/commit/70b5c50b6218521604099a138ad9c48615788611#diff-d1b685eea7a01a36d55534bf94486fceL5

Now instead of trying to read a package.json from a relative location, it's trying to resolve the `pnpm` package and read that package.json. The problem with this is in our projects we use the self-installer to save pnpm into our own specific cache, and so it never gets installed into the node_modules folder it's being run in. This breaks the resolution of `require.resolve(pnpm)`.

### pnpm version: 4.x

### Code to reproduce the issue:
Using the self-installer, install pnpm somewhere
Try getting the pnpm version from `pnpm.js`: `node path/to/lib/bin/pnpm.js -v`

<!--
If there was a fatal error also include a gist of your pnpm-debug.log file.
-->

### Expected behavior:
`node path/to/lib/bin/pnpm.js -v` Outputs the pnpm version

### Actual behavior:
Outputs the following stack trace:

```
(node:98772) UnhandledPromiseRejectionWarning: Error: Cannot find module 'pnpm/package.json'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.resolve (internal/modules/cjs/helpers.js:33:19)
    at Object.<anonymous> (/Users/josephschmitt/development/uc-frontend/.pnpm/4.5.0/lib/node_modules/@pnpm/cli-utils/lib/pnpmPkgJson.js:4:43)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
```

### Additional information:
- Any NodeJS
- macOS 10.15.x

--- a/packages/cli-utils/package.json
+++ b/packages/cli-utils/package.json
@@ -34,7 +34,6 @@
     "@pnpm/types": "workspace:5.0.0",
     "@types/nopt": "^3.0.29",
     "@types/ramda": "^0.26.40",
-    "pnpm": "link:../pnpm",
     "rimraf": "3.0.0"
   },
   "dependencies": {
@@ -51,7 +50,6 @@
     "version-selector-type": "2.0.1"
   },
   "peerDependencies": {
-    "@pnpm/logger": "^3.1.0",
-    "pnpm": "*"
+    "@pnpm/logger": "^3.1.0"
   }
 }

--- a/packages/cli-utils/src/pnpmPkgJson.ts
+++ b/packages/cli-utils/src/pnpmPkgJson.ts
@@ -1,7 +1,10 @@
 import { DependencyManifest } from '@pnpm/types'
 import loadJsonFile = require('load-json-file')
+import path = require('path')
 
-const pkgJson = loadJsonFile.sync<DependencyManifest>(require.resolve('pnpm/package.json', { paths: [__dirname] }))
+const pkgJson = loadJsonFile.sync<DependencyManifest>(
+  path.join(path.dirname(require.main!.filename), '../package.json'),
+)
 const packageManager = {
   name: pkgJson.name,
   // Never a prerelease version

--- a/packages/store-connection-manager/package.json
+++ b/packages/store-connection-manager/package.json
@@ -32,7 +32,6 @@
     "@pnpm/store-connection-manager": "link:",
     "@types/lru-cache": "5.1.0",
     "@types/mz": "^2.7.0",
-    "pnpm": "link:../pnpm",
     "rimraf": "3.0.0"
   },
   "dependencies": {
@@ -52,7 +51,6 @@
     "mz": "2.7.0"
   },
   "peerDependencies": {
-    "@pnpm/logger": "^3.1.0",
-    "pnpm": "*"
+    "@pnpm/logger": "^3.1.0"
   }
 }

--- a/packages/store-connection-manager/src/runServerInBackground.ts
+++ b/packages/store-connection-manager/src/runServerInBackground.ts
@@ -1,6 +1,6 @@
 import diable = require('diable')
 
-const pnpm = require.resolve('pnpm/bin/pnpm.js', { paths: [__dirname] })
+const pnpm = require.main!.filename
 
 export default (storePath: string) => {
   return diable.daemonize(pnpm, ['server', 'start', '--store-dir', storePath], { stdio: 'inherit' })

